test/unit/hooks/set-internal-deadline/index.js

```diff
@@ -0,0 +1,217 @@
+      assert.ok(pplUpdateTask.patch.notCalled);
+    });
+
+    it('adds a 20 working day internal deadline for ppl amendment first submission', () => {
+      const pplAmendmentTask = {
+        data: {
+          model: 'project',
+          action: 'grant',
+          data: {},
+          modelData: {
+            status: 'active'
+          },
+          meta: {}
+        },
+        meta: {
+          payload: {
+            data: {},
+            meta: {}
+          }
+        },
+        activityLog: [],
+        updatedAt: moment().format('YYYY-MM-DD'),
+        patch: sinon.stub()
+      };
+
+      const history = History(pplAmendmentTask);
+      history.status(withInspectorate.id);
+
+      const expected = {
+        internalDeadline: {
+          standard: moment().addWorkingTime(20, 'days').format('YYYY-MM-DD')
+        }
+      };
+
+      runHook(pplAmendmentTask);
+      assert.ok(pplAmendmentTask.patch.calledOnce);
+      assert.deepEqual(pplAmendmentTask.patch.lastCall.args[0], expected);
+    });
+
+    it('adds a 15 working day internal deadline for ppl amendment subsequent submissions', () => {
+      const pplAmendmentResub = {
+        data: {
+          model: 'project',
+          action: 'grant',
+          data: {},
+          modelData: {
+            status: 'active'
+          },
+          meta: {}
+        },
+        meta: {
+          payload: {
+            data: {},
+            meta: {}
+          }
+        },
+        activityLog: [],
+        updatedAt: moment().format('YYYY-MM-DD'),
+        patch: sinon.stub()
+      };
+
+      const history = History(pplAmendmentResub);
+      history.status(withInspectorate.id);
+      history.status(returnedToApplicant.id);
+      history.status(withInspectorate.id);
+
+      const expected = {
+        internalDeadline: {
+          standard: moment().addWorkingTime(15, 'days').format('YYYY-MM-DD')
+        }
+      };
+
+      runHook(pplAmendmentResub);
+      assert.ok(pplAmendmentResub.patch.calledOnce);
+      assert.deepEqual(pplAmendmentResub.patch.lastCall.args[0], expected);
+    });
+
+    it('adds internal deadlines that match statutory deadlines for ppl applications first submission', () => {
+      const pplAmendmentResub = {
+        data: {
+          model: 'project',
+          action: 'grant',
+          data: {},
+          modelData: {
+            status: 'inactive'
+          },
+          meta: {}
+        },
+        meta: {
+          payload: {
+            data: {},
+            meta: {}
+          }
+        },
+        activityLog: [],
+        updatedAt: moment().format('YYYY-MM-DD'),
+        patch: sinon.stub()
+      };
+
+      const history = History(pplAmendmentResub);
+      history.status(withInspectorate.id);
+
+      const expected = {
+        internalDeadline: {
+          standard: moment().addWorkingTime(40, 'days').format('YYYY-MM-DD'),
+          extended: moment().addWorkingTime(55, 'days').format('YYYY-MM-DD')
```

Comment:
We can use the `task.data.deadline.isExtended` flag to decide which of these to use when rendering the deadline, it didn't seem sensible to track/manage the extension in multiple places.